lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(30)
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if ((@back + 1) % @store.length) == @front
+      raise ArgumentError.new "Queue is full"
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if @front == @back
+      raise ArgumentError.new "Queue is full"
+    end
+
+    temp = @store[@front]
+    @store[@front] = nil
+    @front = (@front + 1) % @store.length
```

Comment:
At this point you also need to check to see if `@front == @back` and if so, the queue is empty now and you need to reset them to -1.